### DISCUSSION TITLE
Concern to install the last release

### DISCUSSION THREAD

**Discussion by HOzo123** [flobz/psa_car_controller#340] [Q&A]
Hello,
With the last release of this App, I saw one change on the main file; in the past, the name was "server.py", now it's "__main__.py" and also a change link to the directory.
As usual, I have performed the update of the App (my configuration is a Rapsberry with Debian) with the command "git pull --rebase" and files seems perfectly updated.
To be in accordance with the "main file" of the App, I have change the name of the main file in my "launch file":
this file name is "/usr/local/bin/psa_car_controller" and is an exe.
**Previous:** 
#! /usr/bin/env sh
exec /usr/local/src/psa_car_controller/.env/bin/python3 /usr/local/src/psa_car_controller/server.py "$@"
**New:**
#! /usr/bin/env sh
exec /usr/local/src/psa_car_controller/.env/bin/python3 /usr/local/src/psa_car_controller/psa_car_controller/__main__.py "$@"

and after, I may able to launch the service (using systemd process) as:
ExecStart=/usr/local/src/psa_car_controller/.env/bin/python3 /usr/local/src/psa_car_controller/server.py -f config.json -c charge_config.json --mail xxxx@xx.fr --password xxx

So, when I launch the App with the last update of this App, I have the message as:

pi@raspberrypi:~ $ sudo systemctl status psa_car_controller.service
● psa_car_controller.service - Remote Control of PSA car
   Loaded: loaded (/etc/systemd/system/psa_car_controller.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2022-04-01 19:16:59 CEST; 4s ago
  Process: 8871 ExecStart=/usr/local/bin/psa_car_controller -m -f config.json -c charge_config.json --mail xxxx@xx.fr --password xxxx (code=exited, status=1/FAILURE)
 Main PID: 8871 (code=exited, status=1/FAILURE)

avril 01 19:16:59 raspberrypi systemd[1]: Started Remote Control of PSA car.
avril 01 19:16:59 raspberrypi psa_car_controller[8871]: Traceback (most recent call last):
avril 01 19:16:59 raspberrypi psa_car_controller[8871]:   File "/usr/local/src/psa_car_controller/psa_car_controller/__main__.py", line 13, in <module>
avril 01 19:16:59 raspberrypi psa_car_controller[8871]:     from psa_car_controller.psacc.application.car_controller import PSACarController
avril 01 19:16:59 raspberrypi psa_car_controller[8871]: ModuleNotFoundError: No module named 'psa_car_controller'
avril 01 19:16:59 raspberrypi systemd[1]: psa_car_controller.service: Main process exited, code=exited, status=1/FAILURE
avril 01 19:16:59 raspberrypi systemd[1]: psa_car_controller.service: Failed with result 'exit-code'.

My first idea was link to the librarian 'psa_car_controller' based on the error " raspberrypi psa_car_controller[8871]: ModuleNotFoundError: No module named 'psa_car_controller'
-> but I'm not able to fix it. If you have an Idea to help me ...

Note: I'm using this App for some months, and the first time, I have followed the explaination in "Youtube" attached in the readme.
Thank you in advance for your support.

**Comment by CalicodesIles** [comment]
Same here (Debian on x86 platform)…
Install got broken with the latest psa_car_controller update, knowing that I used the YouTube tutorial first time

**Reply by HOzo123** [reply]
Thank you CalicodesIles; I'm not alone with this concern. It's probably "only" due to wrong link between directory and files...

**Comment by flobz** [comment]
I updated the docs : https://github.com/flobz/psa_car_controller/blob/master/docs/Install.md
Please tell me it's not clear.
I publish the app on pipy to make it easier to install and update.
You shouldn't loose your config, just make sure to execute psa-car-controller in the wright directory.

**Comment by HOzo123** [comment]
Thank you @flobz for your explantions; it's now better but the server seems "not working" - no access at intranet adress "http://127.0.0.1:5000/get_vehicleinfo/VR3UHxxxx as usual.
In fact, I have installed librarian "pip3 install psa-car-controller" in the python environment and when I launch the App, I have warning:
Note: See below the 3 "status requests" not with the same feedback; however, the service is Active.

pi@raspberrypi:~ $ sudo systemctl enable --now psa_car_controller.service
Created symlink /etc/systemd/system/multi-user.target.wants/psa_car_controller.service → /etc/systemd/system/psa_car_controller.service.
pi@raspberrypi:~ $ sudo systemctl status psa_car_controller.service
● psa_car_controller.service - Remote Control of PSA car
   Loaded: loaded (/etc/systemd/system/psa_car_controller.service; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2022-04-03 16:40:03 CEST; 3s ago
 Main PID: 2188 (python3)
    Tasks: 2 (limit: 3720)
   CGroup: /system.slice/psa_car_controller.service
           └─2188 /usr/local/src/psa_car_controller/.env/bin/python3 /usr/local/src/psa_car_controller/psa_car_controller/__main__.py -f config.json -c charge_config.json --mail xxxx@xx.fr --password xxxx

--------------------------------

avril 03 16:40:03 raspberrypi systemd[1]: Started Remote Control of PSA car.
pi@raspberrypi:~ $ sudo systemctl status psa_car_controller.service
● psa_car_controller.service - Remote Control of PSA car
   Loaded: loaded (/etc/systemd/system/psa_car_controller.service; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2022-04-03 16:40:03 CEST; 8s ago
 Main PID: 2188 (python3)
    Tasks: 9 (limit: 3720)
   CGroup: /system.slice/psa_car_controller.service
           └─2188 /usr/local/src/psa_car_controller/.env/bin/python3 /usr/local/src/psa_car_controller/psa_car_controller/__main__.py -f config.json -c charge_config.json --mail xxxx@xx.fr --password xxxx

avril 03 16:40:10 raspberrypi psa_car_controller[2188]:   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
avril 03 16:40:10 raspberrypi psa_car_controller[2188]:   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
avril 03 16:40:10 raspberrypi psa_car_controller[2188]:   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
avril 03 16:40:10 raspberrypi psa_car_controller[2188]:   File "/usr/local/src/psa_car_controller/.env/lib/python3.7/site-packages/psa_car_controller/web/view/views.py", line 399, in <module>
avril 03 16:40:10 raspberrypi psa_car_controller[2188]:     Charging.set_default_price(get_default_car())
avril 03 16:40:10 raspberrypi psa_car_controller[2188]:   File "/usr/local/src/psa_car_controller/.env/lib/python3.7/site-packages/psa_car_controller/psacc/application/charging.py", line 45, in set_default_price
avril 03 16:40:10 raspberrypi psa_car_controller[2188]:     Charging.set_charge_price(charge, conn, cars.get_car_by_vin(charge.vin))
avril 03 16:40:10 raspberrypi psa_car_controller[2188]: AttributeError: 'Car' object has no attribute 'get_car_by_vin'
avril 03 16:40:10 raspberrypi psa_car_controller[2188]: WARNING: dev bundles requested with serve_locally=False.
avril 03 16:40:10 raspberrypi psa_car_controller[2188]: This is not supported, switching to serve_locally=True

----------------------------------------------------

pi@raspberrypi:~ $ sudo systemctl status psa_car_controller.service
● psa_car_controller.service - Remote Control of PSA car
   Loaded: loaded (/etc/systemd/system/psa_car_controller.service; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2022-04-03 16:40:03 CEST; 11min ago
 Main PID: 2188 (python3)
    Tasks: 9 (limit: 3720)
   CGroup: /system.slice/psa_car_controller.service
           └─2188 /usr/local/src/psa_car_controller/.env/bin/python3 /usr/local/src/psa_car_controller/psa_car_controller/__main__.py -f config.json -c charge_config.json --mail xxxx@xx.fr --password xxxx

avril 03 16:40:26 raspberrypi psa_car_controller[2188]: 2022-04-03 16:40:26,285 :: INFO :: mqtt msg received: psa/RemoteServices/events/MPHRTServices/VR3UHZKXxxxxxx b'{"date":"2022-04-03T14:40:19Z","etat_res_elec":0,"precond_state":
avril 03 16:40:26 raspberrypi psa_car_controller[2188]: 2022-04-03 16:40:26,330 :: INFO :: mqtt msg received: psa/RemoteServices/events/MPHRTServices/VR3UHZKXZxxxx b'{"date":"2022-04-03T14:40:25Z","etat_res_elec":0,"precond_state":
avril 03 16:40:26 raspberrypi psa_car_controller[2188]: 2022-04-03 16:40:26,405 :: INFO :: mqtt msg received: psa/RemoteServices/to/cid/AP-ACNT200005894301/VehCharge/state/ProcessManagement b'{"process_date":"2022-04-03T14:40:26Z","vin
avril 03 16:40:27 raspberrypi psa_car_controller[2188]: 2022-04-03 16:40:27,559 :: INFO :: mqtt msg received: psa/RemoteServices/to/cid/AP-ACNT200005894301/VehCharge/state b'{"resp_date":"2022-04-03T14:40:26Z","vin":"VR3UHZKXZxxxxx"
avril 03 16:40:29 raspberrypi psa_car_controller[2188]: 2022-04-03 16:40:29,687 :: INFO :: mqtt msg received: psa/RemoteServices/events/MPHRTServices/VR3UHZKXZKxxxxx b'{"date":"2022-04-03T14:40:29Z","etat_res_elec":0,"precond_state":
avril 03 16:42:08 raspberrypi psa_car_controller[2188]: 2022-04-03 16:42:08,063 :: WARNING :: refresh_vehicle_info: precedent task still alive
avril 03 16:44:08 raspberrypi psa_car_controller[2188]: 2022-04-03 16:44:08,065 :: WARNING :: refresh_vehicle_info: precedent task still alive
avril 03 16:46:08 raspberrypi psa_car_controller[2188]: 2022-04-03 16:46:08,065 :: WARNING :: refresh_vehicle_info: precedent task still alive
avril 03 16:48:08 raspberrypi psa_car_controller[2188]: 2022-04-03 16:48:08,066 :: WARNING :: refresh_vehicle_info: precedent task still alive
avril 03 16:50:08 raspberrypi psa_car_controller[2188]: 2022-04-03 16:50:08,068 :: WARNING :: refresh_vehicle_info: precedent task still alive

Thank you for your support

**Reply by flobz** [reply]
It's a bug related to the latest version, if you don't want to wait please go back to previous one.

**Reply by HOzo123** [reply]
Thank you @flobz !
I hope this concern may help some other users. For sure, I will wait your next release.

**Reply by HOzo123** [reply]
Hello @flobz , I just updated the App some minutes ago; it seems there is still concern(s); the server seems "not working" - no access at intranet adress.
### Warning:
pi@raspberrypi:~ $ sudo systemctl status psa_car_controller.service
● psa_car_controller.service - Remote Control of PSA car
   Loaded: loaded (/etc/systemd/system/psa_car_controller.service; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2022-04-03 17:33:38 CEST; 7s ago
 Main PID: 1358 (python3)
    Tasks: 9 (limit: 3720)
   CGroup: /system.slice/psa_car_controller.service
           └─1358 /usr/local/src/psa_car_controller/.env/bin/python3 /usr/local/src/psa_car_controller/psa_car_controller/__main__.py -f config.json -c charge_config.json --mail xxx@xx.fr --password xxxx

avril 03 17:33:44 raspberrypi psa_car_controller[1358]:   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
avril 03 17:33:44 raspberrypi psa_car_controller[1358]:   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
avril 03 17:33:44 raspberrypi psa_car_controller[1358]:   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
avril 03 17:33:44 raspberrypi psa_car_controller[1358]:   File "/usr/local/src/psa_car_controller/.env/lib/python3.7/site-packages/psa_car_controller/web/view/views.py", line 399, in <module>
avril 03 17:33:44 raspberrypi psa_car_controller[1358]:     Charging.set_default_price(get_default_car())
avril 03 17:33:44 raspberrypi psa_car_controller[1358]:   File "/usr/local/src/psa_car_controller/.env/lib/python3.7/site-packages/psa_car_controller/psacc/application/charging.py", line 45, in set_default_price
avril 03 17:33:44 raspberrypi psa_car_controller[1358]:     Charging.set_charge_price(charge, conn, cars.get_car_by_vin(charge.vin))
avril 03 17:33:44 raspberrypi psa_car_controller[1358]: AttributeError: 'Car' object has no attribute 'get_car_by_vin'
avril 03 17:33:44 raspberrypi psa_car_controller[1358]: WARNING: dev bundles requested with serve_locally=False.
avril 03 17:33:44 raspberrypi psa_car_controller[1358]: This is not supported, switching to serve_locally=True

Thank you for your support

**Reply by lvenditti** [reply]
same issue to me.

WARNING: dev bundles requested with serve_locally=False.
This is not supported, switching to serve_locally=True
2022-04-07 11:12:37,939 :: INFO :: update_data

Some time the server works only from localhost some time from intranet (as per -l 0.0.0.0 option) some time there is an error on web server and does not work at all, nor localhost nor intranet.

Regards

**Reply by flobz** [reply]
This warning can be ignored. You can restart the server with the -d option to see debug log...

**Comment by flobz** [comment]
First of all use the updated [psa-car-controller.service](https://github.com/flobz/psa_car_controller/blob/master/psa-car-controller.service)

**Reply by lvenditti** [reply]
using new .service I have this error:
 Loaded: loaded (/etc/systemd/system/psa_car_controller.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Thu 2022-04-07 11:45:42 CEST; 3s ago
  Process: 28203 ExecStart=/usr/bin/psa-car-controller -c -r -d (code=exited, status=203/EXEC)
 Main PID: 28203 (code=exited, status=203/EXEC)

the '/usr/bin/psa-car-controller' do not exist, also after the command 'pip3 install psa-car-controller'
the command:
'pip3 install psa-car-controller' works fine but after execution I don not find /usr/bin/psa-car-controller.

as test I used this command:
/usr/local/src/psa_car_controller/psa_car_controller/__main__.py -c -r -d 
and the server works fine but only on localhost.

Regards

**Reply by flobz** [reply]
that's normal you have to set -l 0.0.0.0 to listen on all address (like previous version)

**Comment by HOzo123** [comment]
Hello @flobz , I have normally followed your instruction... however, it seems not enough

I have redo "all the process" to update as:
1) update the librarian
sudo apt-get install python3-typing-extensions python3-pandas python3-plotly python3-paho-mqtt  python3-six python3-dateutil python3-brotli  libblas-dev  liblapack-dev gfortran python3-pycryptodome libatlas3-base python3-cryptography

2) Update the librarian in the dedicated environment
pip3 install psa-car-controller

3) Update the App
git pull --rebase

4) Update the launch file as:
in the directory: /usr/local/bin/psa_car_controller
exec /usr/local/src/psa_car_controller/.env/bin/python3 /usr/local/src/psa_car_controller/psa_car_controller/__main__.py "$@"

5) Update the service fiel as:
in the directory /etc/systemd/system/psa_car_controller.service

[Unit]
Description=Remote Control of PSA car
After=network-online.target

[Service]
Type=simple
User=psa_remote
WorkingDirectory=~
ExecStart=/usr/local/bin/psa_car_controller -r -d -f config.json -c charge_config.json --mail xxxx@xxx.fr --password xxxxxx
Restart=on-failure
RestartSec=1m
Environment=PYTHONUNBUFFERED=true

[Install]
WantedBy=multi-user.target

5) And the final result is still the same - no access to adress 127.1.xx
### Log is:
pi@raspberrypi:~ $ sudo systemctl status psa_car_controller.service
● psa_car_controller.service - Remote Control of PSA car
   Loaded: loaded (/etc/systemd/system/psa_car_controller.service; enabled; vendor preset: enabled)
   Active: active (running) since Tue 2022-04-05 19:55:27 CEST; 23s ago
 Main PID: 1617 (python3)
    Tasks: 9 (limit: 3720)
   CGroup: /system.slice/psa_car_controller.service
           └─1617 /usr/local/src/psa_car_controller/.env/bin/python3 /usr/local/src/psa_car_controller/psa_car_controller/__main__.py -r -d -f config.json -c charge_config.json --mail xxxxx@xxxx.fr --password xxxxx

avril 05 19:55:33 raspberrypi psa_car_controller[1617]:   File "<frozen importlib._bootstrap>", line 983, in _find_and_load
avril 05 19:55:33 raspberrypi psa_car_controller[1617]:   File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
avril 05 19:55:33 raspberrypi psa_car_controller[1617]:   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
avril 05 19:55:33 raspberrypi psa_car_controller[1617]:   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
avril 05 19:55:33 raspberrypi psa_car_controller[1617]:   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
avril 05 19:55:33 raspberrypi psa_car_controller[1617]:   File "/usr/local/src/psa_car_controller/.env/lib/python3.7/site-packages/psa_car_controller/web/view/views.py", line 399, in <module>
avril 05 19:55:33 raspberrypi psa_car_controller[1617]:     Charging.set_default_price(get_default_car())
avril 05 19:55:33 raspberrypi psa_car_controller[1617]:   File "/usr/local/src/psa_car_controller/.env/lib/python3.7/site-packages/psa_car_controller/psacc/application/charging.py", line 45, in set_default_price
avril 05 19:55:33 raspberrypi psa_car_controller[1617]:     Charging.set_charge_price(charge, conn, cars.get_car_by_vin(charge.vin))
avril 05 19:55:33 raspberrypi psa_car_controller[1617]: AttributeError: 'Car' object has no attribute 'get_car_by_vin'

**Reply by flobz** [reply]
Can you start this command : pip3 install - U psa-car-controller

**Reply by lvenditti** [reply]
Many thanks!
it worked with -U option.
Now I have /usr/local/bin/psa-car-controller present and the service now starts fine.

Thank a lot for your fantastic job!!

Bye

**Reply by flobz** [reply]
Wonderful! :)

**Reply by HOzo123** [reply]
Hello @flobz 

It's working "as a charme" with the -U option !
Thank you for your strong support and the great job !

**Reply by flobz** [reply]
Your welcome! :)
